streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1228,34 +1226,16 @@ public void cleanUp() {
      * @return map of store names to another map of partition to {@link LagInfo}s
      */
     public Map<String, Map<Integer, LagInfo>> allLocalStorePartitionLags() {
-        final long latestSentinel = -2L;
         final Map<String, Map<Integer, LagInfo>> localStorePartitionLags = new TreeMap<>();
-
         final Collection<TopicPartition> allPartitions = new LinkedList<>();
         final Map<TopicPartition, Long> allChangelogPositions = new HashMap<>();
 
         // Obtain the current positions, of all the active-restoring and standby tasks
         for (final StreamThread streamThread : threads) {
-            for (final StandbyTask standbyTask : streamThread.allStandbyTasks()) {
```

Comment:
Here we return the allTasks and remove the restoringTaskId, instead we check if the returned position from `changelogOffsets` are Task.LATEST_OFFSET or not at the end. Would rely on your impl inside StreamTask to return that value.